docs/aws/audit/opensearchmonitoring/rules/opensearch_in_vpc_only.mdx

```diff
@@ -22,6 +22,258 @@ CISAWS,HIPAA,ISO27001,RBI_MD_ITF,RBI_UCB
+
+2. List all OpenSearch domains: Use the `aws es list-domain-names` command to list all the OpenSearch (formerly known as Elasticsearch) domains. This command will return a list of all the domain names.
+
+   Command: 
+   ```
+   aws es list-domain-names
+   ```
+   This will return a list of domain names in the following format:
+   ```
+   {
+       "DomainNames": [
+           {
+               "DomainName": "example-domain"
+           },
+           ...
+       ]
+   }
+   ```
+
+3. Describe the OpenSearch domain: For each domain, use the `aws es describe-elasticsearch-domain --domain-name <DomainName>` command to get detailed information about the domain.
+
+   Command: 
+   ```
+   aws es describe-elasticsearch-domain --domain-name example-domain
+   ```
+   This will return a JSON object with detailed information about the domain.
+
+4. Check if the domain is in a VPC: In the output of the previous command, look for the `VPCOptions` field. If this field is present and the `VPCId` field is not null, then the OpenSearch domain is in a VPC. If the `VPCOptions` field is not present or the `VPCId` field is null, then the OpenSearch domain is not in a VPC.
+
+   Example of a domain in a VPC:
+   ```
+   "VPCOptions": {
+       "VPCId": "vpc-1a2b3c4d",
+       ...
+   }
+   ```
+   Example of a domain not in a VPC:
+   ```
+   "VPCOptions": {
+       "VPCId": null,
+       ...
+   }
+   ```
+   or
+   ```
+   "VPCOptions": {}
+   ```
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. This can be done using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Establish a session: You need to establish a session using your AWS credentials. Replace 'aws_access_key_id', 'aws_secret_access_key', and 'region_name' with your AWS credentials.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. List all OpenSearch domains: Use the OpenSearch client to list all the OpenSearch domains and check if they are inside a VPC.
+
+   ```python
+   es_client = session.client('es')
+
+   response = es_client.list_domain_names()
+
+   for domain in response['DomainNames']:
+       domain_name = domain['DomainName']
+       domain_details = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+       if 'VPCOptions' not in domain_details['DomainStatus']:
+           print(f"Domain {domain_name} is not in a VPC")
+   ```
+
+4. Analyze the output: The script will print the names of all OpenSearch domains that are not inside a VPC. If no such domains exist, it will not print anything. This way, you can easily detect if OpenSearch is in a VPC or not.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>